src/core/credentials/transport/xds/xds_credentials.cc

```diff
@@ -156,9 +197,26 @@ XdsCredentials::create_security_connector(
         }
       }
       tls_credentials_options->set_verify_server_cert(true);
+      if (XdsSniEnabled()) {
+        auto maybe_hostname = args->GetOwnedString(GRPC_ARG_ADDRESS_NAME);
+        if (xds_certificate_provider->auto_host_sni() &&
+            maybe_hostname.has_value()) {
+          tls_credentials_options->set_sni_override(maybe_hostname.value());
+        } else if (xds_certificate_provider->sni().length() > 0) {
+          tls_credentials_options->set_sni_override(
+              xds_certificate_provider->sni());
+        } else {
+          if (UseChannelAuthorityIfNoSNIApplicable()) {
+            tls_credentials_options->set_sni_override(std::nullopt);
```

Comment:
The default is nullopt, so I don't think we actually have to set anything in this case.  Instead, we can just use:

```
if (!UseChannelAuthorityIfNoSNIApplicable()) {
  tls_credentials_options->set_sni_override("");
}
```